driver-sync/src/test/functional/com/mongodb/client/unified/ValueMatcher.java

```diff
@@ -44,9 +45,10 @@ public void assertValuesMatch(final BsonValue expected, final BsonValue actual)
         assertValuesMatch(expected, actual, null, -1);
     }
 
-    private void assertValuesMatch(final BsonValue initialExpected, @Nullable final BsonValue actual,
+    private void assertValuesMatch(final BsonValue initialExpected, @Nullable final BsonValue initialActual,
                                    @Nullable final String keyContext, final int arrayPositionContext) {
         BsonValue expected = initialExpected;
+        BsonValue actual = initialActual == null ? BsonNull.VALUE : initialActual;
```

Comment:
This to handle the first occurrence of `expectedResult: null` in a spec test.

Reply:
I removed that line, and also removed handling for `$$exists` in the same switch statement, as test runs show that the case is not entered, and the assertions in the block don't make sense if `actual` is never null.

Reply:
After running the full test suite against more variants, it's clear that `$$exists` is still needed here.  I changed the logic in a different way to handle `actual == null` which I think makes more sense.